src/io/futures_io.rs

```diff
@@ -0,0 +1,183 @@
+//! Compatibility between the `tokio::io` and `futures-io` versions of the
+//! `AsyncRead` and `AsyncWrite` traits.
+use pin_project::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A compatibility layer that allows conversion between the
+/// `tokio::io` and `futures-io` `AsyncRead` and `AsyncWrite` traits.
+#[pin_project]
+#[derive(Copy, Clone, Debug)]
+pub struct Compat<T> {
+    #[pin]
+    inner: T,
+}
+
+/// Extension trait that allows converting a type implementing
+/// `futures_io::AsyncRead` to implement `tokio::io::AsyncRead`.
+pub trait FuturesAsyncReadCompatExt: futures_io::AsyncRead {
+    /// Wraps `self` with a compatibility layer that implements
+    /// `tokio_io::AsyncWrite`.
+    fn compat(self) -> Compat<Self>
```

Comment:
name this compat_read?

Reply:
@LucioFranco the naming scheme was based on `futures_util`'s `io` compat, which has an [`AsyncRead::compat`](https://github.com/rust-lang-nursery/futures-rs/blob/4fef913048cc083f824a5cc4d22d2beb435bb936/futures-util/src/io/mod.rs#L370-L375) and an [`AsyncWrite::compat_write`](https://github.com/rust-lang-nursery/futures-rs/blob/4fef913048cc083f824a5cc4d22d2beb435bb936/futures-util/src/io/mod.rs#L463-L471)

Reply:
although, I realize now that these names should probably be renamed, since they conflict with the same names as `futures-util`'s `AsyncReadExt`/`AsyncWriteExt` traits...previously, the compatibility stuff in `futures-util` was defined in separate ext traits, so that wasn't an issue.

WDYT of `as_tokio`/`as_futures`?

Reply:
(cc @carllerche for API naming)

Reply:
those sound good to me.